packages/react-instantsearch-dom/src/components/InfiniteHits.js

```diff
@@ -62,7 +62,11 @@ InfiniteHits.propTypes = {
   refineNext: PropTypes.func.isRequired,
   translate: PropTypes.func.isRequired,
   className: PropTypes.string,
-  hitComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  hitComponent: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.func,
+    PropTypes.object,
```

Comment:
Why is it `PropTypes.object` for infiniteHits and `PropTypes.any` in hits?

Reply:
object isn't actually the right type (it needs to be a specific object with $$typeof), which isn't describable in PropTypes as well as TypeScript at the same time. As the prop types aren't really that important in this case, I went with the strictest way that still worked with both TS in hits, but a slightly stricter way in the .js one. 

As tried in a previous commit, the real right way is using elementType, but it increased the bundle size significantly

Reply:
so why not put the same (though non ideal) type for both, just to keep it consistent? Is it possible?

Reply:
sure, they could both be `any`